CONTRIBUTING.md

```diff
@@ -30,103 +73,115 @@ $ git clone https://github.com/exercism/interactive-cli-walkthrough
 
-### Issues
+Files should be saved with the `*.tw2` extension.
 
-We keep track of bugs, enhancements and support requests in the repository using GitHub [issues][].
+##### Including files
 
-### Pull Requests
+The main file which includes everything is [main.tw2][main.tw2]. In order for
+your file to appear in the walkthrough, [main.tw2][main.tw2] should contain the
+path to your file.
 
-When submitting a pull request, sometimes we'll ask you to make changes before
-we accept the patch.
+#### Additional commands
 
-Please do not close the first pull request and open a second one with these
-changes. If you push more commits to a branch that you've opened a pull
-request for, it automatically updates the pull request.
+To make it easy to create links between steps, two special commands are
+available:
 
-When the pull request is ready to be merged, you will probably be asked to [squash your commits][squash-commits].
+##### Step title
 
-As with adding more commits, you do not need to close your pull request and open a new one.
-If you change the history (rebase, squash, amend), and use git push --force to update the branch on your fork.
-The pull request points to that branch, not to specific commits.
+`::Link Destination`
 
-Here's a guide on [how to squash commits in a GitHub pull request][squash-commits].
+This sets a step's name as `Link Destination` which is needed in order for the
+link command to know what step to link to. This command is placed on the first
+line of the file.
 
-When submitting a PR, please make sure to include the compiled HTML and the
-newly generated instruction graph in order for your changes to appear on the
-live site.
+##### Link
+
+`[[<Link Text>-><Link Destination>]]`
 
-#### Compiling to HTML
+This creates a link with `Link Text` that links to a step entitled
+`Link Destination`.
 
-In order to compile to HTML, run the following command:
+#### Viewing changes
+
+In order to view your changes, you need to compile the project. Compiling is
+done by running this command:
 
 ```
 $ twee2 build main.tw2 compiled.html --format=Snowman
 ```
 
-#### Generating the instruction graph
+Compiling produces a file named `compiled.html`. Open this file in your browser
+and you should see your changes.
+
+#### Visualizing steps
 
-In order to generate the instruction graph, run the following command:
+If you're a bit lost of how each step links to another, you could generate
+the walkthrough graph by running this command:
 
 ```
 $ rake graph
 ```
 
-### Development workflow
+This produces a file named `graph.png`. Opening this file shows you
+how each step links to another. This tool is also helpful to find broken links
+between steps.
 
-The walkthrough is written in Markdown. In order to get a hang of the syntax,
-please read this [tutorial][markdown].
+### Submitting Code Changes
 
-#### Additional commands
+These instructions should get you closer to getting a commit into the
+repository.
```

Comment:
How about:

> If you're new to contributing to open source on GitHub, this next section should help you get started. If you get stuck, open an issue to ask us for help and we'll get you sorted out (and improve these instructions).

That helps frame this as _details that you may or may not need depending on your experience with GitHub_, and also emphasizes that we are happy to help, and that if a person is confused it's probably our fault, not theirs.

Reply:
Great idea! We should change the exercism contributing guide to use this statement, too.

Reply:
Agreed!